macos/Sources/Features/Terminal/TerminalView.swift

```diff
@@ -86,32 +86,32 @@ struct TerminalView<ViewModel: TerminalViewModel>: View {
                 if (Ghostty.info.mode == GHOSTTY_BUILD_MODE_DEBUG || Ghostty.info.mode == GHOSTTY_BUILD_MODE_RELEASE_SAFE) {
                     DebugBuildWarningView()
                 }
-
+                
                 Ghostty.TerminalSplit(node: $viewModel.surfaceTree)
-                    .environmentObject(ghostty)
+                    .environment(ghostty)
                     .focused($focused)
                     .onAppear { self.focused = true }
-                    .onChange(of: focusedSurface) { newValue in
-                        self.delegate?.focusedSurfaceDidChange(to: newValue)
+                    .onChange(of: focusedSurface) {
+                        self.delegate?.focusedSurfaceDidChange(to: focusedSurface)
                     }
-                    .onChange(of: title) { newValue in
-                        self.delegate?.titleDidChange(to: newValue)
+                    .onChange(of: title) {
+                        self.delegate?.titleDidChange(to: title)
                     }
-                    .onChange(of: pwdURL) { newValue in
-                        self.delegate?.pwdDidChange(to: newValue)
+                    .onChange(of: pwdURL) {
+                        self.delegate?.pwdDidChange(to: pwdURL)
                     }
-                    .onChange(of: cellSize) { newValue in
-                        guard let size = newValue else { return }
+                    .onChange(of: cellSize) {
+                        guard let size = cellSize else { return }
                         self.delegate?.cellSizeDidChange(to: size)
                     }
-                    .onChange(of: viewModel.surfaceTree?.hashValue) { _ in
+                    .onChange(of: viewModel.surfaceTree?.hashValue) {
```

Comment:
Given that `Container` is now `@Observable`, this may no longer be required. A view will observe the properties of all `@Observable` instances it reads from during `body`, which means we no longer need to manually propagate changes up from child objects. I am not confident enough to remove this, however.